userspace/libscap/linux/scap_procs.c

```diff
@@ -501,11 +501,11 @@ int32_t scap_proc_fill_cgroups(char* error, int cgroup_version, struct scap_thre
 
 int32_t scap_proc_fill_pidns_start_ts(char* error, struct scap_threadinfo* tinfo, const char* procdirname)
 {
-	char filename[SCAP_MAX_PATH_SIZE];
+	char proc_cmdline_pidns[SCAP_MAX_PATH_SIZE];
 	struct stat targetstat = {0};
 
-	snprintf(filename, sizeof(filename), "%sroot/proc/1", procdirname);
-	if(stat(filename, &targetstat) == 0)
+	snprintf(proc_cmdline_pidns, sizeof(proc_cmdline_pidns), "%sroot/proc/1/cmdline", procdirname);
+	if(stat(proc_cmdline_pidns, &targetstat) == 0)
```

Comment:
Just a note: with this implementation, the "container start time" for host processes will not be equal to the boot time but (presumably) to the time when the host init started.

I am completely fine with this (and it does make sense), just pointing this out since we spent the whole Friday discussing the subtle differences of various timestamps :)

Reply:
@gnosek hmmm having problems with `if (tinfo->vpid != tinfo->pid)` to ever evaluate to true even when placing the check after those fields should have been parsed. Then also never got to `else if(strstr(line, "NStgid:") == line)` in `scap_proc_fill_info_from_stats`, so in summary yes it should be a follow up PR, something with that parser seems wonky as well!

Reply:
@incertum can you show me the code?

Reply:
Just place that comparison if statement anywhere after vpid should have been populated or even add print or debug statements within that scap_proc_fill_info_from_stats parser, it never evaluated to true for me even though I was running containers etc with sleep processes. Is it working for you?